### DISCUSSION TITLE
Question about querying nested ManyToMany relation with filter

### DISCUSSION THREAD

**Discussion by sorasful** [collerek/ormar#506] [Q&A] [answered]
Hello there ! 
I'm continuing my learning curve of Ormar and I'm at a point where I have some troubles with M2M relation and filtering. I tried to search in the documentation, issues and discussions but I did not find an answer to my question. 

TL;DR : I'm trying to get some results by searching using a specific value in a M2M relationship.

I have 3 models :  Playlist (can contain multiple Sources: M2M) -> Source (Can be in multiple playlists: M2M) and (Can have multiple Entries but entries can have ONE source, so it's just a foreign key) -> Entries 

Here are the models so you can check everything's okay: 

```python
class Playlist(ormar.Model):
   """ Can have multiple sources"""
    class Meta(BaseMeta):
        tablename = 'playlists'

    id: int = ormar.Integer(primary_key=True)
    title: str = ormar.String(max_length=50, nullable=False, index=True)

class Source(ormar.Model):
   """Can be in multiple playlists"""
    class Meta(BaseMeta):
        tablename = 'sources'

    id: int = ormar.Integer(primary_key=True)
    title: str = ormar.String(max_length=400, index=True, nullable=True)
    
    playlists: Optional[List['Playlist']] = ormar.ManyToMany(Playlist)

class Entry(ormar.Model):
    """ Entries are attached to ONE source """
    class Meta(BaseMeta):
        tablename = 'entries'

    id: int = ormar.Integer(primary_key=True)
    title: str = ormar.String(max_length=300, index=True, nullable=False)
    
    source: Optional['Source'] = ormar.ForeignKey(to=Source, related_name="entries")

```

So, in my unit test, i'm populating one of my source with entries : 

```python
    # adding new entries
    for i in range(5):
        entry = Entry(
            title=f"Entry{i}",
            link=f"https://{i}.com",
        )
        await playlist.sources[0].entries.add(entry)

    await playlist.update()

```

And now is the problem :  

If I query directly my playlist, I got this :  
![image](https://user-images.githubusercontent.com/32820423/148257054-8f7ecb79-64f5-44e1-9f57-34f568cf3649.png)

I also checked that the entries are well saved :  
![image](https://user-images.githubusercontent.com/32820423/148256883-ecb806a4-1f65-456b-9339-477efbf2d452.png)

And If I try with the "correct" way, it's giving me empty array, always. I tried to play with select_related, fields ... But no luck for now .. 
![image](https://user-images.githubusercontent.com/32820423/148256944-fcfa7236-cf4a-4ad2-8cf2-55f98d19b130.png)

I'm sorry for the noob question, but I'm sure this can help other people (hi future me). I thank you for your time and help :)

**Comment by collerek** [comment, marked as answer]
That seems correct at first look. 
You don't need to update playlist after adding related items to m2m.

I'm missing the part when you save the playlist itself and the sources, can you share this too?

**Reply by sorasful** [reply]
Okay, so I feel very dumb, you're absolutely right, it's correct. It's just that in my fixture I kept adding the sources to my user user .append instead of .add (I'm moving from sync SQLalchemy). 

I was too much focused on this I did not check the above. Thank you very much for your time, that's kind of you to answer so quickly.
